File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    
    def hanoiTop(n, s, e, by):
        if n == 1:
            answer.append([s,e])
            return
        
        hanoiTop(n-1, s, by, e);
        answer.append([s, e]);
        hanoiTop(n-1, by, e, s);
        
    hanoiTop(n, 1, 3, 2)
    
    return answer

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
  const answer = [];
    
  const hanoiTop = (n, s, e, by) => {
    if (n === 1) {
      answer.push([s, e]);
      return;
    }
      
    hanoiTop(n - 1, s, by, e);
    answer.push([s, e]);
    hanoiTop(n - 1, by, e, s);
  }
  
  hanoiTop(n, 1, 3, 2);
    
  return answer;
}